src/mlpack/bindings/cli/CMakeLists.txt

```diff
@@ -50,11 +50,20 @@ if (BUILD_CLI_EXECUTABLES)
   add_executable(mlpack_${name}
     ${name}_main.cpp
   )
+
+  if(CMAKE_CROSSCOMPILING)
```

Comment:
Should we use `BUILD_SHARED_LIBS` as the variable to check here instead?

Reply:
I do not mind if it gives the same results

Reply:
I think that would be better, or alternately perhaps an additional variable to set if the user wants statically-compiled programs too.  We may also want to consider `CMAKE_FIND_LIBRARY_SUFFIXES` too: https://blog.kitware.com/creating-static-executables-on-linux/ (since whenever making a static program all dependencies must be statically linked too).